modules/cli/src/main/scala/scala/cli/commands/WatchUtil.scala

```diff
@@ -9,18 +11,47 @@ object WatchUtil {
   def waitMessage(message: String): String = {
     // Both short cuts actually always work, but Ctrl+C also exits mill in mill watch mode.
     val shortCut = if (isDevMode) "Ctrl+D" else "Ctrl+C"
-    val gray     = "\u001b[90m"
-    val reset    = Console.RESET
-    s"$gray$message, press $shortCut to exit, or press Enter to re-run.$reset"
+    gray(s"$message, press $shortCut to exit, or press Enter to re-run.")
+  }
+
+  private def gray(message: String): String = {
+    val gray  = "\u001b[90m"
+    val reset = Console.RESET
+    s"$gray$message$reset"
   }
 
   def printWatchMessage(): Unit =
     System.err.println(waitMessage("Watching sources"))
 
-  def waitForCtrlC(onPressEnter: () => Unit = () => ()): Unit = {
+  def printWatchWhileRunningMessage(): Unit =
+    System.err.println(gray("Watching sources while your program is running."))
```

Comment:
NIT I'm thinking the logger could be passed here and used to print the watch messages, to respect verbosity... 
But it's not a big deal, can be done separately.